File: Minimum_Depth_of_Binary_Tree.cpp
/**
 * Definition of TreeNode:
 * class TreeNode {
 * public:
 *     int val;
 *     TreeNode *left, *right;
 *     TreeNode(int val) {
 *         this->val = val;
 *         this->left = this->right = NULL;
 *     }
 * }
 */
class Solution {
public:
    /**
     * @param root: The root of binary tree.
     * @return: An integer
     */
    int minDepth(TreeNode *root) {
        // write your code here
        if (!root)
        {
            return 0;
        }
        if (root->left && root->right)
        {
            return 1 + min(minDepth(root->left), minDepth(root->right));
        }
        else
        {
            return 1 + max(minDepth(root->left), minDepth(root->right));
        }
    }
};

File: Minimum Depth of Binary Tree.py
"""
Definition of TreeNode:
class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left, self.right = None, None
"""
class Solution:
    """
    @param root: The root of binary tree.
    @return: An integer
    """ 
    def minDepth(self, root):
        # write your code here
        
        def helper(node):
            if not node:
                return 0
            
            if not node.left and not node.right:
                return 1
            elif not node.left or not node.right:
                return 1 + max(helper(node.left), helper(node.right))
            else:
                return 1 + min(helper(node.left), helper(node.right))
        
        return helper(root)
                